File: eccorce_app/app/src/main/java/com/example/eccomerce_app/dto/OrderDto.kt
package com.example.eccomerce_app.dto

import com.example.hotel_mobile.services.kSerializeChanger.UUIDKserialize
import com.example.hotel_mobile.services.kSerializeChanger.UUIDListKserialize
import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
data class OrderDto(
    @Serializable(with = UUIDKserialize::class)
    val Id: UUID,
    val Longitude: Double,
    val Latitude: Double,
    val TotalPrice: Double,
    val DeliveryFee: Double,
    val UserPhone: String,
    val Status:Int,
    val OrderItems:List<OrderItemDto>
)

@Serializable
data class CreateOrderDto(
    val Longitude: Double,
    val Latitude: Double,
    val TotalPrice: Double,
    val Items: List<CreateOrderItemDto>,
)
@Serializable
data class OrderRequestItemsDto(
    @Serializable(with = UUIDKserialize::class)
    val StoreId: UUID,
    val Price: Double,
    val Quantity: Int,
    @Serializable(with = UUIDKserialize::class)
    val ProductId: UUID,
    @Serializable(with = UUIDListKserialize::class)
    val ProductsVarientId: List<UUID>,
)

File: backend/api/Presentation/dto/OrderDto.cs
using System.ComponentModel.DataAnnotations;

namespace api.Presentation.dto
{
    public class OrderDto
    {
        public Guid Id { get; set; }
        public decimal Longitude { get; set; }
        public decimal Latitude { get; set; }
        public decimal TotalPrice { get; set; }
        public decimal DeliveryFee { get; set; }
        public String Name { get; set; }
        public String UserPhone { get; set; }
        public string Status { get; set; }
        public List<OrderItemDto>? OrderItems { get; set; } = null;
    }

    public class AdminOrderDto
    {
        public List<OrderDto>? Orders { get; set; }
        public int pageNum { get; set; } = 1;
    }
    public class DeliveryOrderDto
    {
        public Guid Id { get; set; }
        public decimal Longitude { get; set; }
        public decimal Latitude { get; set; }
        public decimal TotalPrice { get; set; }
        public decimal RealPrice { get; set; }
        public decimal DeliveryFee { get; set; }
        public String Name { get; set; }
        public String UserPhone { get; set; }
        public int Status { get; set; }
        public List<DeliveryOrderItemDto> OrderItems { get; set; }
    }

    public class CreateOrderDto
    {
        [Required] public decimal Longitude { get; set; }
        [Required] public decimal Latitude { get; set; }
        [Required] public decimal TotalPrice { get; set; }
        [Required] public List<CreateOrderItemDto> Items { get; set; }
    }

    public class OrderTookByEvent
    {
        public Guid Id { get; set; }
        public Guid DeliveryId { get; set; }
    }

    public class UpdateOrderStatusDto
    {
        public Guid Id { get; set; }
        public int Status { get; } 
    }
    
    public class UpdateOrderStatusEventDto
    {
        public Guid Id { get; set; }
        public string Status { get; set; } 
    }
}